tests/test_dem.py

```diff
@@ -18,39 +23,75 @@
 
 class TestDEM:
 
-    def test_load_DEM_subclass(self):
+    def test_load(self):
+
+        # check that the loading from DEM __init__ does not fail
+        fn_img = os.path.join(path_module,EXAMPLE_PATHS['dem1'])
+        img = DEM(fn_img)
 
-        fn_img = EXAMPLE_PATHS['dem1']
+    def test_set_vref(self):
 
+        fn_img = os.path.join(path_module,EXAMPLE_PATHS['dem1'])
         img = DEM(fn_img)
 
-    def test_vref(self):
+        # check for WGS84
+        img.set_vref(vref_name='WGS84')
+        assert img.vref == 'WGS84'
+        assert img.vref_grid is None
+
+        # check for EGM96
+        img.set_vref(vref_name='EGM96')
+        assert img.vref == 'EGM96'
+        assert img.vref_grid == 'us_nga_egm96_15.tif'
+        # grid should have priority over name and parse the right vref name
+        img.set_vref(vref_name='WGS84',vref_grid='us_nga_egm96_15.tif')
+        assert img.vref == 'EGM96'
+
+        # check for EGM08
+        img.set_vref(vref_name='EGM08')
+        assert img.vref == 'EGM08'
+        assert img.vref_grid == 'us_nga_egm08_25.tif'
+        # grid should have priority over name and parse the right vref name
+        img.set_vref(vref_name='best ref in the entire world, or any string',vref_grid='us_nga_egm08_25.tif')
+        assert img.vref == 'EGM08'
+
+        # check that other existing grids are well detected in the pyproj.datadir
+        img.set_vref(vref_grid='is_lmi_Icegeoid_ISN93.tif')
+
+        # check that non-existing grids raise errors
+        with pytest.raises(ValueError):
+            img.set_vref(vref_grid='the best grid in the entire world, or any non-existing string')
 
-        #somewhere in Chile
+
+    def test_to_vref(self):
+
+        #first, some points to test the transform
+
+        # Chile
         lat = 43.70012234
         lng = -79.41629234
         z = 100
-
-        #egm1996
-
         with warnings.catch_warnings():
             warnings.filterwarnings("ignore", module="pyproj")
             #init is deprecated by
             ellipsoid=pyproj.Proj(init="EPSG:4326") #WGS84 datum ellipsoid height
             geoid=pyproj.Proj(init="EPSG:4326", geoidgrids='egm96_15.gtx') #EGM96 geoid in Chile, we expect ~30 m difference
-
         transformer = pyproj.Transformer.from_proj(ellipsoid, geoid)
-        print(transformer.transform(lng,lat,z))
+        z_out = transformer.transform(lng,lat,z)[2]
+
+        assert np.logical_and.reduce((np.isfinite(z_out),np.greater(z_out,z),np.less(np.abs(z_out-z),40)))
```

Comment:
what's the meaning of this? #CommentsAreGreat ;-)

Reply:
Added :D

Reply:
So why don't you test that it is exactly the value that you currently get?